plugins/extractors/kafka/kafka.go

```diff
@@ -114,6 +150,36 @@ func (e *Extractor) Extract(ctx context.Context, emit plugins.Emit) (err error)
 	return
 }
 
+func (e *Extractor) createTLSConfig() (tlsConfig *tls.Config, err error) {
+	tlsConfig = &tls.Config{
+		InsecureSkipVerify: e.config.Auth.TLS.InsecureSkipVerify,
+	}
+
+	authConfig := e.config.Auth.TLS
+	if *authConfig.CertFile != "" && *authConfig.KeyFile != "" && *authConfig.CAFile != "" {
+		cert, err := tls.LoadX509KeyPair(*authConfig.CertFile, *authConfig.KeyFile)
+		if err != nil {
+			return nil, errors.Wrap(err, "unable to create cert")
+		}
+
+		caCert, err := os.ReadFile(*authConfig.CAFile)
+		if err != nil {
+			return nil, errors.Wrap(err, "unable to read ca cert file")
+		}
+
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
+
+		tlsConfig = &tls.Config{
+			Certificates:       []tls.Certificate{cert},
+			RootCAs:            caCertPool,
+			InsecureSkipVerify: e.config.Auth.TLS.InsecureSkipVerify,
+		}
+	}
+
+	return tlsConfig, nil
+}
```

Comment:
nit: use early return and align the happy path to the left. Blog post elaborating on this - https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

```suggestion
func (e *Extractor) createTLSConfig() (*tls.Config, error) {
	authConfig := e.config.Auth.TLS
	if *authConfig.CertFile == "" || *authConfig.KeyFile == "" || *authConfig.CAFile == "" {
		return &tls.Config{
			InsecureSkipVerify: e.config.Auth.TLS.InsecureSkipVerify,
		}, nil
	}

	cert, err := tls.LoadX509KeyPair(*authConfig.CertFile, *authConfig.KeyFile)
	if err != nil {
		return nil, errors.Wrap(err, "unable to create cert")
	}

	caCert, err := os.ReadFile(*authConfig.CAFile)
	if err != nil {
		return nil, errors.Wrap(err, "unable to read ca cert file")
	}

	caCertPool := x509.NewCertPool()
	caCertPool.AppendCertsFromPEM(caCert)

	return &tls.Config{
		Certificates:       []tls.Certificate{cert},
		RootCAs:            caCertPool,
		InsecureSkipVerify: e.config.Auth.TLS.InsecureSkipVerify,
	}, nil
}
```